clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -1494,6 +1494,27 @@ private static void assertEmptyPendingRequests(CommitRequestManager commitReques
         assertTrue(commitRequestManager.pendingRequests.unsentOffsetCommits.isEmpty());
     }
 
+    @Test
+    public void testPollWithFatalErrorDuringCoordinatorIsEmptyAndClosing() {
```

Comment:
This unit test only covers the path of `coordinatorRequestManager.coordinator().isEmpty() -> pendingRequests.maybeFailOnCoordinatorFatalError`. Could you please add another one to cover the case of `closing && pendingRequests.hasUnsentRequests()`?